ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -120,7 +120,7 @@ public XAResourceRecord(TransactionImple tx, XAResource res, Xid xid,
 
 		if (params != null)
 		{
-			if (params.length >= XACONNECTION)
+			if (params.length >= XACONNECTION + 1)
```

Comment:
Well I am not sure how clever the compiler but I think that it will perform the addition and then do the compare which is a few more instructions that a single compare. So yes change it in both places.